.env.template

```diff
@@ -0,0 +1,5 @@
+# openai keys
+OPENAI_API_KEY=
```

Comment:
what do you think is the advantage for using env compared with using export ENV_NAME.

For env, maybe add something like OPENAI_API_KEY="xxx"?

Reply:
> what do you think is the advantage for using env compared with using export ENV_NAME.



Hey @lwaekfjlk ,
When we integrate redis(or some other vectorDB), they generally have their set of ENV variables we need to provide.
This is similar to the scenario when we want to use any other llm provider(such as azure etc) apart from default openai in litellm.

So, to accommodate the increasing number of env variables, it'll be easier to set the env variables by sourcing the env file rather than individually setting each variable. In shell, the command to source the env file looks something like this:
```bash
$ set -a
$ source .env
$ set +a
```

Let me know your thoughts on this.

Reply:
> For env, maybe add something like OPENAI_API_KEY="xxx"?

The idea behind this is, that `.env.template` file is a placeholder file which specify all the env variable names the project is using. Users will create a `.env` file based on this template(they can just copy the entire template file).

Putting empty value denotes that the variable wasn't set by the user and can be handled appropriately in python code. Something like
```python
api_key = os.getenv("OPENAI_API_KEY") # given empty string if variable is not set
asset api_key # Assertion error if API_KEY is not set, similar to not setting the value
```

Let me know your thoughts.